hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/parallelnotification/client/ParallelNotificationConnector.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2020 Bosch.IO GmbH and others.
+*
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*/
+package org.eclipse.hawkbit.ui.common.parallelnotification.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.hawkbit.ui.common.parallelnotification.ParallelNotification;
+
+import com.google.gwt.dom.client.Style.Unit;
+import com.vaadin.client.ServerConnector;
+import com.vaadin.client.ui.VNotification;
+import com.vaadin.client.ui.notification.NotificationConnector;
+import com.vaadin.shared.ui.Connect;
+import com.vaadin.shared.ui.notification.NotificationServerRpc;
+import com.vaadin.shared.ui.notification.NotificationState;
+
+/**
+ * Connector for {@link ParallelNotification} that holds all active
+ * notifications of the current UI and positions them so they do not overlap.
+ */
+@Connect(value = ParallelNotification.class)
+public class ParallelNotificationConnector extends NotificationConnector {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Map<VNotification, Integer> notifications = new HashMap<>();
```

Comment:
Please double check if it is OK to use a class variable here. This should be scoped on the current UI (not be shared between browser tabs). It seems to work like that, but maybe there is a better solution.